support/android/apk/buildSrc/src/main/kotlin/Interop.kt

```diff
@@ -0,0 +1,67 @@
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+import java.io.File
+import java.util.Locale
+import java.util.Properties
+
+fun Project.getTargetDir(debug: Boolean, arch: String): String {
+    val basePath = project.rootDir.parentFile.parentFile.parentFile.absolutePath
+    return basePath + "/target/android/" + getSubTargetDir(debug, arch)
+}
+
+fun Project.getNativeTargetDir(debug: Boolean, arch: String): String {
+    val basePath = project.rootDir.parentFile.parentFile.parentFile.absolutePath
+    return basePath + "/target/" + getSubTargetDir(debug, arch)
+}
+
+fun getSubTargetDir(debug: Boolean, arch: String): String {
+    return getRustTarget(arch) + "/" + if (debug) "debug" else "release"
+}
+
+fun Project.getJniLibsPath(debug: Boolean, arch: String): String =
```

Comment:
Why are some functions defined on the Project but not others?

Reply:
Unlike in groovy in which scope is implicit and not clear, Kotlin requires explicit scoping.

`Project.` means this function is an extension to `Project` classes, thus has access to public fields of the `Project` class, and can only be called from a project object or scope.

I shall add documentation to the top of the file in this regard.

Reply:
[786edc0](https://github.com/servo/servo/pull/33772/commits/786edc07ad24676396fc98815c44e7dc70d8c879)